pkg/cmd/repo/create/create_test.go

```diff
@@ -0,0 +1,255 @@
+package create
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/cli/cli/internal/config"
+	"github.com/cli/cli/internal/run"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/httpmock"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/test"
+	"github.com/google/shlex"
+	"github.com/stretchr/testify/assert"
+)
+
+func runCommand(httpClient *http.Client, cli string) (*test.CmdOut, error) {
+	io, stdin, stdout, stderr := iostreams.Test()
+	fac := &cmdutil.Factory{
+		IOStreams: io,
+		HttpClient: func() (*http.Client, error) {
+			return httpClient, nil
+		},
+		Config: func() (config.Config, error) {
+			return config.NewBlankConfig(), nil
+		},
+	}
+
+	cmd := NewCmdCreate(fac, nil)
+
+	// TODO STUPID HACK
+	// cobra aggressively adds help to all commands. since we're not running through the root command
+	// (which manages help when running for real) and since create has a '-h' flag (for homepage),
+	// cobra blows up when it tried to add a help flag and -h is already in use. This hack adds a
+	// dummy help flag with a random shorthand to get around this.
+	cmd.Flags().BoolP("help", "x", false, "")
+
+	argv, err := shlex.Split(cli)
+	cmd.SetArgs(argv)
+
+	cmd.SetIn(stdin)
+	cmd.SetOut(stdout)
+	cmd.SetErr(stderr)
```

Comment:
Could this be set to throwaway Buffer instances instead of to the real IO streams?

This would be so we avoid actually writing using Cobra's `cmd.Print*` utils and instead enforce always writing through the iostreams instance.